src/security/builtin_plugins/tests/encode_datawriter_submessage/src/encode_datawriter_submessage_utests.c

```diff
@@ -557,97 +556,73 @@ static bool crypto_decrypt_data(uint32_t session_id, unsigned char *iv, DDS_Secu
 
   /* create the cipher context */
   ctx = EVP_CIPHER_CTX_new();
-  if (ctx)
+  if (!ctx)
   {
-    if (key_size == 128)
-    {
-      if (!EVP_DecryptInit_ex(ctx, EVP_aes_128_gcm(), NULL, NULL, NULL))
-      {
-        ERR_print_errors_fp(stderr);
-        result = false;
-      }
-    }
-    else if (key_size == 256)
-    {
-      if (!EVP_DecryptInit_ex(ctx, EVP_aes_256_gcm(), NULL, NULL, NULL))
-      {
-        ERR_print_errors_fp(stderr);
-        result = false;
-      }
-    }
+    ERR_print_errors_fp(stderr);
+    return false;
   }
 
-  if (result)
+  if (key_size == 128 && !EVP_DecryptInit_ex(ctx, EVP_aes_128_gcm(), NULL, NULL, NULL))
   {
-    if (!EVP_DecryptInit_ex(ctx, NULL, NULL, session_key.data, iv))
+    ERR_print_errors_fp(stderr);
+    goto err;
+  }
+  if (key_size == 256 && !EVP_DecryptInit_ex(ctx, EVP_aes_256_gcm(), NULL, NULL, NULL))
+  {
+    ERR_print_errors_fp(stderr);
+    goto err;
```

Comment:
Maybe just nitpicking, but what if `key_size` is neither `128` or `256`? We could opt to make it an assert(?) Also, the `ERR_print_errors_fp` is always combined with `goto err;`, you could choose to just call `ERR_print_errors_fp` from the `err:` label... I'm fine with it like this though.

Reply:
I've added the assert on invalid `key_size` in 7709a3e. I considered moving the `ERR_print_errors_fp` call, but I think that  causes the reported line number in the error message to be the `err:` block and not the line where the error occurred.